### PR TITLE
Added documentation required to show component error properly

### PR DESCRIPTION
**Pull request by TheMutsi**
This is a new section in the react native Docs, to guide user to implement the config metro needed to show component error properly

### CODE REVIEW

File: docs/react-native/features/tracking-error-component.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: Tracking Render Errors
+description: Ensure meaningful screen names in render errors by configuring Metro bundler.
+sidebar_position: 7
+---
+
+# Tracking Render Errors
+
+To accurately track React Native rendering errors and include meaningful screen names in the `componentStack`, you need to configure your `metro.config.js` file. By default, in release builds, screen names may appear as `Unknown`. The following configuration ensures that class and function names are preserved during minification.
```
Comment (jpmunz) [inline docs/react-native/features/tracking-error-component.md:9]: I'd maybe rephrase to make it a bit more clear it's optional and that we do still log something by default, maybe:


> If the Embrace SDK detects a component stack trace in a React Native rendering error it will log it automatically. By default, in release builds, screen names may appear as `Unknown` as in the following example:
-->example from below

> It is possible to configure these names to be preserved during minification to make it easier to debug rendering errors in release builds at the cost of a slightly larger bundle size:
-->example from below

File: docs/react-native/features/tracking-error-component.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+This configuration helps retain screen names in the `componentStack`, making it easier to debug rendering errors in release builds.
+
+## Important Note
+
+This configuration **could increase the size of your JavaScript bundle**. Although in our tests we did not notice any significant difference in the bundle size, we strongly recommend testing it in your project to confirm.
+
+## Checking the Bundle Size
+
+To analyze the bundle size, run the following command:
```
Comment (jpmunz) [inline docs/react-native/features/tracking-error-component.md:47]:
```suggestion
## Checking the Bundle Size

We strongly recommend testing your project to confirm that any added bundle size due to this configuration is acceptable. To analyze the bundle size, run the following command:
```

File: docs/react-native/features/tracking-error-component.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+Replace `<platform>` with either `android` or `ios`. This command generates the bundle and assets, allowing you to compare the size with and without the Metro configuration.
+
+## Example of `componentStack`
+
+Below is an example of how the `componentStack` appears with and without the Metro configuration:
+
+### With Metro Configuration:
+```text
+ * at undefined (in Screen2)
```
Comment (jpmunz) [inline docs/react-native/features/tracking-error-component.md:66]: will these always be `at undefined` ? Are there examples where this has more useful information?

File: docs/react-native/features/tracking-error-component.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+module.exports = mergeConfig(getDefaultConfig(__dirname), config);
+```
+
+## Explanation
+
+- **`keep_classnames: true` and `keep_fnames: true`**: These options ensure that class names and function names are preserved during the minification process.  
+- **`mangle`**: Prevents the renaming of class and function names.  
+
+This configuration helps retain screen names in the `componentStack`, making it easier to debug rendering errors in release builds.
```
Comment (jpmunz) [inline docs/react-native/features/tracking-error-component.md:39]: Moved reasoning above, also think it can be in the same section as the code snippet

```suggestion

- **`keep_classnames: true` and `keep_fnames: true`**: These options ensure that class names and function names are preserved during the minification process.  
- **`mangle`**: Prevents the renaming of class and function names.  
```

File: docs/react-native/features/tracking-error-component.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+title: Tracking Render Errors
+description: Ensure meaningful screen names in render errors by configuring Metro bundler.
+sidebar_position: 7
+---
+
+# Tracking Render Errors
+
+To accurately track React Native rendering errors and include meaningful screen names in the `componentStack`, you need to configure your `metro.config.js` file. By default, in release builds, screen names may appear as `Unknown`. The following configuration ensures that class and function names are preserved during minification.
+
+## Implementation
```
Comment (jpmunz) [inline docs/react-native/features/tracking-error-component.md:11]:
```suggestion
## Preserving screen names
```

File: docs/react-native/features/tracking-error-component.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+Add the following configuration to your `metro.config.js` file:
+
+```javascript
+const {getDefaultConfig, mergeConfig} = require('@react-native/metro-config');
+
+const config = {
+  transformer: {
+    minifierConfig: {
+      keep_classnames: true,
```
Comment (jpmunz) [inline docs/react-native/features/tracking-error-component.md:21]: are both these needed? If the app only uses functional components is `keep_classnames` still required?

**Review by jpmunz — COMMENTED** [review]
`features/index.md` should be updated as well to have this entry in its list

File: docs/react-native/features/tracking-error-component.md (hunk 3 above)
Comment (TheMutsi) [reply docs/react-native/features/tracking-error-component.md]: We decided to remove at undefined and the parentheses

**Review by facostaembrace — APPROVED** [review]
